Python/pylifecycle.c

```diff
@@ -1911,6 +1911,9 @@ Py_FinalizeEx(void)
     int malloc_stats = tstate->interp->config.malloc_stats;
 #endif
 
+    /* Ensure that remaining threads are detached */
+    _PyEval_StopTheWorldAll(runtime);
```

Comment:
Does PyEnd_Interpreter() need a similar call?

Reply:
No, because `PyEnd_Interpreter()` doesn't handle the case of remaining threads.

I think we ought to do a better job of handling ctrl-c in `PyEnd_Interpreter()`, but not in this PR.